crypto/ecdsa.go

```diff
@@ -67,6 +67,15 @@ func ECDSAKeyPairFromKey(priv *ecdsa.PrivateKey) (PrivKey, PubKey, error) {
 	return &ECDSAPrivateKey{priv}, &ECDSAPublicKey{&priv.PublicKey}, nil
 }
 
+// ECDSAPublicKeyFromPubKey generates a new ecdsa public key from an input public key
+func ECDSAPublicKeyFromPubKey(pub *ecdsa.PublicKey) (PubKey, error) {
```

Comment:
From some conversation during triage:

It is likely better to dereference they key before it's passed in to reduce the likelihood of hard to track down bugs. Also the keys aren't too big